### ISSUE TITLE
API reference?

### ISSUE THREAD

**Issue by laurent22** [andpor/react-native-sqlite-storage#163]
Is there an API reference for this library? For example, I see many different `openDatabase` calls in the README file. How can I find out what are all the possible parameters and values?

**Comment by ewendel** [comment]
I've been looking for this as well. @andpor?

**Comment by rodrigoelp** [comment]
Once the typescript definition file gets in, you can use it as documentation for your js files as well.

**Comment by andpor** [comment]
Look at documentation on openDatabase call - it shows various options for locations and states where target database file gets created...

**Comment by palerdot** [comment]
@andpor Actually, I think there should be a documentation on how to use the library for various use cases like `SELECT/INSERT/CREATE` etc. Right now, I'm looking around but there is no information on how exactly to use this library. I can find instructions on only how to set up.

**Comment by rodrigoelp** [comment]
@palerdot, I think the documentation you are looking for is the same as the [cordova SQLite project](https://github.com/litehelpers/Cordova-sqlite-storage)

SQLite works pretty much the same regardless of the plugin adapting the library.

I got working examples [here](https://github.com/rodrigoelp/reactnative-typescript-exercise-15/blob/master/src/index.tsx) if you are interested. It should read pretty similar to pure javascript as typescript is javascript with types.

By the way, I got the typescript definition to work on my project about a week or so ago. I see the pull request created by @dryganets #220 has not been merged. Is there something wrong with his definition? I am curious to know if there is a problem with his as our implementations are pretty similar with minor differences.

**Comment by palerdot** [comment]
@rodrigoelp The examples you shared is for typescript. Is there any normal JS/ES6 examples. I myself started compiling by digging into the code samples here - https://github.com/palerdot/react-native-sqlite-storage-examples. Also, this api information should go into the README of this repo. As most people, will not look for documentation in another repo.

**Comment by rodrigoelp** [comment]
Hi @palerdot, I do not have any js/es6 sample handy. If I got some time soon I will try to generate the same code in javascript.

**Comment by giantss** [comment]
I think that should provide API documentation, so it looks more intuitive.

**Comment by MillingtonAllen** [comment]
> Also, this api information should go into the README of this repo. As most people, will not look for documentation in another repo.

Yeah...  Links in the readme to the API reference (even if other github repo) and the example repos would be real nice.
